### PR TITLE
Feature/67 assessment widget tray api

### LINKED ISSUE #67: Assessment Widget Tray configuration API
There should an assessment widget tray configuration API. It should:

- [x] use the data types:
  ```ts
  type AssessmentWidgetConfig = {
    dimensionEh: EntryHash,
    widgetEh?: EntryHash // This is specifically for when components are separated out into their own DHT entry (or sequence of DHT entries to allow extra large codebases to be stored).
  } | {
    dimensionEh: EntryHash,
    applicationId?: EntryHash, // This is whatever the id for the application is. I didn't check to see if this is accurate
    componentName?: string // This is the name of the component as exposed by the applet interface
  }
  type AssessmentWidgetBlockConfig = {
    // This is the widget that allows making an assessment and displaying the user's chosen selection if the user can select one of many options
    inputAssessmentWidget: AssessmentWidgetConfig,
    outputAssessmentWidget: {
        dimensionEh: EntryHash,
        widgetEh: EntryHash
        // There is currently no agreement about having multiple components, one for assessment and one for pure output of computed assessments.
        // However, if there were separate components, the config for that component would go here and this would be the same as above
        // There is also a discussion to be had about other modalities of sensemaking like tags and things we haven't considered yet.
    }
  }
  // I'm torn between making this an ordered hash or just an array. I think for simplicity, this will be an array.
  type AssessmentWidgetTrayConfig = AssessmentWidgetBlockConfig[]
  type ResourceDefAssessmentWidgetTrayConfig = {
    [ResourceDefEH]: AssessmentWidgetTrayConfig
  }
  ```
- [x] have a method to set or replace `AssessmentWidgetTrayConfig` for a given `ResourceDefEh`.
- [x] have a method to retrieve an `AssessmentWidgetTrayConfig` given a `ResourceDefEh`.

### PR DESCRIPTION
**Pull request by pospi**
Implements #67.

### CODE REVIEW

**Comment by adaburrows** [comment]
Please see the comments I left on the original ticket. The simplifications you made don't support the use cases in the UI.

**Comment by pospi** [comment]
should be good to go.

File: dnas/sensemaker/zomes/integrity/widgets/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,154 @@
+use hdi::prelude::*;
+use sensemaker_integrity_structs::Properties;
+
+#[derive(Debug, Clone, Serialize, Deserialize, SerializedBytes)]
+#[serde(rename_all = "camelCase")]
+pub struct DimensionWidgetBinding {
```
Comment (adaburrows) [inline dnas/sensemaker/zomes/integrity/widgets/src/lib.rs:6]: Rename to `DimensionStandaloneWidgetBinding`

File: dnas/sensemaker/zomes/integrity/widgets/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    pub dimension_eh: EntryHash,
+    // id of the widget CustomElement stored in companion zome
+    pub widget_eh: EntryHash,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, SerializedBytes)]
+#[serde(rename_all = "camelCase")]
+pub struct DimensionAppletFeatureBinding {
```
Comment (adaburrows) [inline dnas/sensemaker/zomes/integrity/widgets/src/lib.rs:14]: Rename to `DimensionAppletWidgetBinding`

**Review by adaburrows — APPROVED** [review]
I'm going to create a ticket for the changes I want done and the client changes that weren't done in this PR. But merging since otherwise it'll block people.

**Pull request merged**